### PR TITLE
Use jsonnet to generate the manifests

### PR DESCRIPTION
**Pull request by wanghaoran1988**

### CODE REVIEW

File: jsonnet/openshift-state-metrics.libsonnet [hunk 1]
```diff
@@ -0,0 +1,301 @@
...
+
+
+      local routeRule = rulesType.new() +
+                         rulesType.withApiGroups(['route.openshift.io']) +
+                         rulesType.withResources([
+                           'routes',
+                         ]) +
+                         rulesType.withVerbs(['list', 'watch']);
+
+      local authenticationRole = rulesType.new() +
```
Comment (aditya-konarde) [inline jsonnet/openshift-state-metrics.libsonnet]: Nit: Should this be called `authenticationRule` for consistency with others?

File: jsonnet/openshift-state-metrics.libsonnet [hunk 2]
```diff
@@ -0,0 +1,301 @@
...
+                                 rulesType.withApiGroups(['authentication.k8s.io']) +
+                                 rulesType.withResources([
+                                   'tokenreviews',
+                                 ]) +
+                                 rulesType.withVerbs(['create']);
+
+      local authorizationRole = rulesType.new() +
```
Comment (aditya-konarde) [inline jsonnet/openshift-state-metrics.libsonnet]: Nit:Same as above Should this be called `authorizationRule`?

File: jsonnet/openshift-state-metrics.libsonnet [hunk 3]
```diff
@@ -0,0 +1,301 @@
+local k = import 'ksonnet/ksonnet.beta.3/k.libsonnet';
```
[the remark this answers is not in the document]
Comment (wanghaoran1988) [reply jsonnet/openshift-state-metrics.libsonnet]: Let's use beta.3 to align with kube-state-metrics.

[the remark this answers is not in the document]
Comment (wanghaoran1988) [reply jsonnet/openshift-state-metrics.libsonnet]: ok, will update later

**Pull request merged**